src/test/java/com/cloudant/tests/ViewsTest.java

```diff
@@ -573,25 +799,138 @@ private Page checkPagesBackward(int currentPage, int numberOfPages, Page page, b
+        json.addProperty("created", 1020);
+        json.addProperty("boolean", true);
+        json.addProperty("total", 14.34);
+        json.addProperty("quotes", "\"quotes\"");
+        json.addProperty("spaces", " space ");
+        db.save(json);
+
+        DesignDocument designDoc = new DesignDocument();
+
+        designDoc.setId("_design/keyarray_example");
+        designDoc.setLanguage("javascript");
+
+
+        DesignDocument.MapReduce mapFunctions =
+                new DesignDocument.MapReduce();
+
+        //Creator and created multi (3) value key map function
+        mapFunctions.setMap(
+                "function(doc) { emit([doc.boolean, doc.creator, doc.created], null); }");
+
+
+        Map<String, DesignDocument.MapReduce> keyViews =
+                new HashMap<String, DesignDocument.MapReduce>();
+        keyViews.put("creator_created_boolean", mapFunctions);
+
+        designDoc.setViews(keyViews);
+
+        //Created and spaces multi (2) value key map function
+        mapFunctions =
+                new DesignDocument.MapReduce();
+
+        mapFunctions.setMap(
+                "function(doc) { emit([doc.spaces, doc.created], null); }");
+
+        keyViews.put("spaces_created", mapFunctions);
+
+        designDoc.setViews(keyViews);
+
+        //Created and quotes multi (2) value key map function
+        mapFunctions =
+                new DesignDocument.MapReduce();
+
+        mapFunctions.setMap(
+                "function(doc) { emit([doc.quotes, doc.created], null); }");
+
+        keyViews.put("quotes_created", mapFunctions);
+
+        designDoc.setViews(keyViews);
+
+        //Creator and created multi (2) value key map function
+        mapFunctions =
+                new DesignDocument.MapReduce();
+
+        mapFunctions.setMap(
+                "function(doc) { emit([doc.creator, doc.created], null); }");
+
+        keyViews.put("creator_created", mapFunctions);
+
+        designDoc.setViews(keyViews);
+
+        //Created and total multi (2) integer value key map function
+        mapFunctions =
+                new DesignDocument.MapReduce();
+
+        mapFunctions.setMap(
+                "function(doc) { emit([doc.created, doc.total], null); }");
+
+        keyViews.put("created_total", mapFunctions);
+
+        designDoc.setViews(keyViews);
+
+        //Created single value map function for integer key test
+        mapFunctions =
+                new DesignDocument.MapReduce();
+
+        mapFunctions.setMap(
+                "function(doc) { emit([doc.created], null); }");
+
+        keyViews.put("created", mapFunctions);
+
+        designDoc.setViews(keyViews);
+
+        //Created single value map function for boolean key test
+        mapFunctions = new DesignDocument.MapReduce();
+
+        mapFunctions.setMap(
+                "function(doc) { emit([doc.boolean], null); }");
+
+        keyViews.put("boolean", mapFunctions);
+
+        designDoc.setViews(keyViews);
```

Comment:
I was going to create map.js classes under the resources/example/views folder.  This is where the views for the rest of the test cases (like example/foo) exist.